### ISSUE TITLE
Serverless v4 issue with plugin getting objects as parameters

### ISSUE THREAD

**Issue by alexandre-walzberg** [serverless/serverless#12815]
### Issue description

Hello,

My plugin is broken when using serverless v4, allthough it is claimed there are no breaking changes in v4.
In contrary to serverless v3, it seems it does not accept parameter that are objects (I mean entire yml blocks).

plugin: https://www.npmjs.com/package/@ynpact/serverless-plugin-foreach

Here is the issue with v4, at packaging step (same template in v3 does not cause any issue) 👍 
```
✖ Failed to resolve placeholder '${self:custom.templateRepeat}' at path 'resources.0'. String value consist of variable which resolve with non-string value.
Error: Failed to resolve placeholder '${self:custom.templateRepeat}' at path 'resources.0'. String value consist of variable which resolve with non-string value.
    at #replacePlaceholderInConfig (file:///home/apalepex/.serverless/releases/4.3.3/package/dist/sf-core.js:655:57996)
    at #handlePlaceholderNode (file:///home/apalepex/.serverless/releases/4.3.3/package/dist/sf-core.js:655:53296)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async file:///home/apalepex/.serverless/releases/4.3.3/package/dist/sf-core.js:655:51233
    at async processNodeAndHandleCompletion (file:///home/apalepex/.serverless/releases/4.3.3/package/dist/sf-core.js:655:42013)
```

**sls version** command output : 4.3.3
**node version** : 21.4.0
**npm version** : 10.3.0

The same does not produce error using serverless v3.

Note : if I copy/paste the custom.templateRepeat, custom.templateForeach and custom.bucketParams into separate yaml file and use the ${file()} source variable as parameter of ${foreach} and ${repeat}, it works.

### Context

serverless.yml
```
service: service

plugins:
  - '@ynpact/serverless-plugin-foreach'
custom:
  templateRepeat:
    bucket{{i}}:
      Type: AWS::S3::Bucket
      Properties:
        BucketName: bucket_{{i}}
  templateForeach:
    bucketFE{{i}}:
      Type: AWS::S3::Bucket
      Properties:
        BucketName: "{{i.name}}"
        PublicAccessBlockConfiguration:
          BlockPublicAcls: "{{i.blockPublicAcls}}"
  bucketParams:
    - name: my-public-bucket
      blockPublicAcls: false
    - name: my-private-bucket
      blockPublicAcls: true

provider:
  name: aws
  region: eu-west-1
  runtime: nodejs18.x
  stage: ${opt:stage, 'dev'}
  environment:
    AWS_NODEJS_CONNECTION_REUSE_ENABLED: 1
    LOG_LEVEL: "debug"


resources:
  - ${repeat(${self:custom.templateRepeat}, 3)}
  - ${foreach(${self:custom.templateForeach}, ${self:custom.bucketParams})}
```
package.json (note: plugin is linked via npm link) : 
```
{
  "name": "sample_app",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "esbuild": "^0.20.2",
    "serverless": "^4.3.3",
    "serverless-esbuild": "^1.52.1",
    "typescript": "^5.5.3"
  }
}
```

**Comment by czubocha** [comment]
Hey @alexandre-walzberg, thanks for reporting. We're looking into it.

**Comment by czubocha** [comment]
Hey @alexandre-walzberg, this has been addressed in version 4.4.3, which is now live. Feel free to try it out and let us know if you run into any issues.
